scripts/docker-helper-4.x/cht-docker-compose.sh

```diff
@@ -140,12 +154,91 @@ required_apps_installed(){
 }
 
 get_nginx_container_id() {
-	docker ps --all --filter "publish=${NGINX_HTTPS_PORT}" --filter "name=${projectName}" --quiet
+  docker ps --all --filter "publish=${NGINX_HTTPS_PORT}" --filter "name=${projectName}" --quiet  2>/dev/null
 }
 
 get_is_container_running() {
-	containerId=$1
-	docker inspect --format="{{.State.Running}}" "$containerId" 2>/dev/null
+  containerId=$1
+  docker inspect --format="{{.State.Running}}" "$containerId" 2>/dev/null
+}
+
+service_has_image_downloaded(){
+	service=$1
+  if [ "$service" == "cht-upgrade-service" ]; then
+    compose_path="${homeDir}/upgrade-service.yml"
+	elif [ "$service" == "couchdb" ]; then
+    compose_path="${homeDir}/compose/couchdb.yml"
+	else
+    compose_path="${homeDir}/compose/cht-core.yml"
+	fi
+	image=$(grep "${service}:" ${compose_path} | grep image | cut -f2,3 -d":" | xargs)
+
+  imageDownloadName=$(docker image ls  --format {{.Repository}}:{{.Tag}} -f "reference=${image}" 2>/dev/null)
+	if [ $imageDownloadName ];then
+	  echo ${imageDownloadName}
+  else
+    echo "NA"
+  fi
+}
+
+service_has_container(){
+	service=$1
+  container_name=$(docker ps -af "name=^${projectName}[-_]+.*[-_]+[0-9]" --format '{{.Names}}' | grep ${service} 2>/dev/null)
+  if [ $container_name ];then
+    echo ${container_name}
+  else
+    echo "NA"
+  fi
+}
+
+container_status(){
+	contianer=$1
+	status=$(docker inspect --format="{{.State.Status}}" "$contianer" 2>/dev/null)
+  if [ $status ];then
+    echo ${status}
+  else
+    echo "NA"
+  fi
+}
```

Comment:
There's a little indentation weirdness here - mixing tabs and spaces maybe?

Reply:
yikes! Thanks for the catch - fixed in b41a52d79